src/main/webapp/WEB-INF/pages/productList.jsp

```diff
@@ -8,20 +8,28 @@
   <p>
     Welcome to Expert-Soft training!
   </p>
+  <form>
+      <input name = "query" value = ${param.query}>
+      <button>Search</button>
+  </form>
   <table>
     <thead>
       <tr>
         <td>Image</td>
-        <td>Description</td>
-        <td class="price">Price</td>
+        <td class ="description">Description <a href ="products?sort=description&Board=asc&query =${param.query}">asc</a>
+            <a href ="products?sort=description&Board=desc&query =${param.query}">desc</a> 
+        </td>
+        <td class="price">Price <a href = "products?sort=price&Board=asc&query =${param.query}">asc</a>
+            <a href = "products?sort=price&Board=desc&query =${param.query}">desc</a>
+        </td>
       </tr>
     </thead>
     <c:forEach var="product" items="${products}">
       <tr>
         <td>
           <img class="product-tile" src="https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/${product.imageUrl}">
         </td>
-        <td>${product.description}</td>
+        <td> <a href = "products/description?prod=${product.id}">${product.description}</a></td>
         <td class="price">
```

Comment:
Price history popup here?

Task "Clicking product price shall open a popup with price change history."

Reply:
This task is selfcheck and I thought it wasn't necessary to do by the first check